gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/UnArchiveConverter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.converter;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.Converter;
+import gobblin.converter.DataConversionException;
+import gobblin.converter.SchemaConversionException;
+import gobblin.converter.SingleRecordIterable;
+import gobblin.data.management.copy.FileAwareInputStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+
+
+/**
+ * A {@link Converter} that converts an archived {@link InputStream} to a tar {@link InputStream}. Wraps the given
+ * archived {@link InputStream} with {@link GZIPInputStream} Use this converter if the {@link InputStream} from source
+ * is compressed.
+ */
+public class UnArchiveConverter extends Converter<String, String, FileAwareInputStream, FileAwareInputStream> {
```

Comment:
This looks more like an UnGzipConverter or InflaterConverter, not an UnArchiveConverter. Can you also add support for both gzip and Zip to make it more generic? For example, you can add a config option to specify the type of inflater algorithm:
* GZIP
* ZIP
*AUTO_BY_NAME (default)

AUTO_BY_NAME can check the extension and use GZIP for .gz and .tgz and ZIP for .zip.